### ISSUE TITLE
'rails server' command line does not work with Thin when SSL is enabled

### ISSUE THREAD

**Issue by tkrotoff** [macournoyer/thin#218]
tl;dr Thin should be able to replace WEBrick in a totally transparent manner, this is not the case when SSL is enabled.

See original [Rails issue](https://github.com/rails/rails/issues/11822), does not seem to be Rails related after all: [explanations](https://github.com/rails/rails/issues/11822#issuecomment-34591045).

After replacing WEBrick by Thin (`gem 'thin'` inside my Gemfile), command line `rails server` should work.

It does indeed but only when Rails is not configured in SSL mode.
If configured with SSL, Thin outputs `!! Invalid request` for every HTTPS request on the server:

```
$ rails server
=> Booting Thin
=> Rails 3.2.14 application starting in development on https://0.0.0.0:3000
=> Call with -d to detach
=> Ctrl-C to shutdown server
>> Thin web server (v1.5.1 codename Straight Razor)
>> Maximum connections set to 1024
>> Listening on 0.0.0.0:3000, CTRL+C to stop
!! Invalid request
```

If I run Thin directly using `thin start --ssl` then it works fine and I don't have any error.

Here is my `script/rails`

``` Ruby
#!/usr/bin/env ruby.exe

require 'rails/commands/server'
require 'webrick/https'

module Rails
  class Server < ::Rack::Server
    def default_options
      super.merge({
        SSLEnable: true,
        SSLPrivateKey: OpenSSL::PKey::RSA.new(File.open('ssl/bfo_app.cert.key').read),
        SSLCertificate: OpenSSL::X509::Certificate.new(File.open('ssl/bfo_app.cert.crt').read)
      })
    end
  end
end

APP_PATH = File.expand_path('../../config/application',  __FILE__)
require File.expand_path('../../config/boot',  __FILE__)
require 'rails/commands'
```

Inside application.rb I've also set `config.force_ssl = true`

Tested using Rails 3.2.14 under OS X and Win7

Why it is interesting to use `rails server` instead of `thin start --ssl`?

When using the command line this is not a problem, it's when using an IDE like RubyMine that this becomes interesting.
RubyMine calls `rails server` with a list of arguments and there is no way to tell RubyMine to call `thin start --ssl` instead:

![RubyMine 5.4 run configuration window](http://s9.postimg.org/noq19andb/Ruby_Mine_run_configuration.png)

(screenshot from RubyMine latest version: 5.4 - Win7)
